src/main/java/org/asamk/signal/Manager.java

```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
         sendMessage(messageBuilder, recipients);
     }
 
+    @Override
+    public String getContactName(String number) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            return number;
```

Comment:
I think it's better to return an empty string here, to indicate, that the contact is unknown